pkg/clusterversion/cv.go

```diff
@@ -221,6 +222,42 @@ func GetCurrentVersion(clusterVersion *configv1.ClusterVersion) (string, error)
 	return gotVersion, nil
 }
 
+// GetCurrentVersionMinusOne strings a latest version -1 as a string and error
+func GetCurrentVersionMinusOne(clusterVersion *configv1.ClusterVersion) (string, error) {
+	var gotVersionMinusOne string
+	var completedTimes []*metav1.Time
+
+	for _, history := range clusterVersion.Status.History {
+		if history.State == configv1.CompletedUpdate {
+			completedTimes = append(completedTimes, history.CompletionTime)
+		}
+	}
+
+	if len(completedTimes) <= 1 {
+		return gotVersionMinusOne, fmt.Errorf("cluster has only one version available")
```

Comment:
Question - If a cluster is freshly installed, its expected that it would have only single version. When MUO would run on it, we would expect it to not error during reconcile since there's only single version. Should we return an error here?

Reply:
Fair point. the collector only calls this func if the UpgradeConfig desired.version = ClusterVersion desired.version.https://github.com/openshift/managed-upgrade-operator/pull/287/files#diff-82dc411a0ed86675e9efddaa43dadc4fc6a75ea0e672400429e8ea39424da44cR321 
This is the failsafe for atleast this flow. If this function was called else where, the caller could check the error string and choose to continue based on the error returned.